packages/core/src/components/Box/react/box.test.tsx

```diff
@@ -17,4 +17,13 @@ describe("Box component", () => {
       color: "rgb(255, 255, 255)",
     });
   });
+
+  it("should correctly override prop types by `as` prop", () => {
+    // Arrange
+    const ref = createRef<HTMLButtonElement>();
+    const onClick: MouseEventHandler<HTMLButtonElement> = () => {};
+
+    // Assert
+    <Box as="button" ref={ref} onClick={onClick} />;
```

Comment:
@poteboy Hmm it seems that the type of `<Box ... />` is Element but not ComponentProps.

Reply:
Right, but I think you can grasp what I meant to say. Testing the type by using `expectTypeOf` is semantically more accurate isn't it? (Even if it doesn’t change the test result.)

Reply:
Yes, it's definitely better if we can use `expectTypeOf`. But I don't know how to extract the prop type to test, which is dynamically changed by `as` prop

Reply:
Ah, I see. 
When using `expectTypeOf`, the test would look like the following.
However, this test is just testing the details of the types and focus solely on testing the intricacies of the types.
From the point of view of meaningful testing, it seems a bit questionable... 🤔

```ts
expectTypeOf<ComponentProps<"button">>().toMatchTypeOf<ComponentProps<typeof Box<"button">>>();
```

It may be better not to use `expectTypeOf` in this case.